norace_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !race
+
+package nats
+
+import (
+	"testing"
+	"time"
+)
+
+func TestParseStateReconnectFunctionality(t *testing.T) {
```

Comment:
If you move this to a new "norace_test.go" file, I would recommend naming this test TestNoRace... so that from travis you add a new command to run all the "no race" tests like: `go test -v -run=TestNoRace -p=1 ./...`

Reply:
Sounds reasonable.